ui/src/alerting/components/CheckHistoryVisualization.tsx

```diff
@@ -0,0 +1,78 @@
+// Libraries
+import React, {FC, createContext, useState} from 'react'
+import {get} from 'lodash'
+
+// Components
+import {Plot} from '@influxdata/giraffe'
+import CheckPlot from 'src/shared/components/CheckPlot'
+import EmptyQueryView, {ErrorFormat} from 'src/shared/components/EmptyQueryView'
+
+// Types
+import {Check, TimeZone, CheckViewProperties} from 'src/types'
+import TimeSeries from 'src/shared/components/TimeSeries'
+import {createView} from 'src/shared/utils/view'
+import {checkResultsLength} from 'src/shared/utils/vis'
+import {getTimeRangeVars} from 'src/variables/utils/getTimeRangeVars'
+import {TimeRange} from 'src/types'
+
+interface ResourceIDs {
+  checkIDs: {[x: string]: boolean}
+  endpointIDs: {[x: string]: boolean}
+  ruleIDs: {[x: string]: boolean}
+}
+
+export const ResourceIDsContext = createContext<ResourceIDs>(null)
+
+interface OwnProps {
+  check: Check
+  timeZone: TimeZone
+}
+
+type Props = OwnProps
+//TODO maybe update submitToken when we know how
+
+const CheckHistoryVisualization: FC<Props> = ({check, timeZone}) => {
+  const view = createView<CheckViewProperties>(get(check, 'type', 'threshold'))
```

Comment:
I think we ought to be able to assume check has type here.

Reply:
Removed!